api/src/test/java/com/epam/pipeline/manager/pipeline/RunScheduleManagerTest.java

```diff
@@ -103,214 +117,238 @@ public void setUp() {
+                .when(mockPipelineRunManager).loadAllRunsByPipeline(testPipeline.getId());
 
-        final List<RunSchedule> loadRunSchedule = runScheduleManager
-                .loadAllSchedulesBySchedulableId(RUN_ID, ScheduleType.PIPELINE_RUN);
-        assertEquals(0, loadRunSchedule.size());
-        Mockito.verify(runScheduler, Mockito.times(2)).unscheduleRunSchedule(any());
+        runScheduleManager.deleteSchedulesForRunByPipeline(testPipeline.getId());
 
+        verify(mockRunScheduler, times(2)).unscheduleRunSchedule(any());
+        verify(mockRunScheduleDao, times(1)).deleteRunSchedulesForRunByPipeline(anyLong());
     }
 
     @Test(expected = IllegalArgumentException.class)
-    @WithMockUser(username = USER_OWNER)
     public void loadRunScheduleByWrongId() {
-        final RunSchedule runSchedule =
-            runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
-                    Collections.singletonList(testRunScheduleVO)).get(0);
-        runScheduleManager.loadSchedule(runSchedule.getSchedulableId());
+        doReturn(Optional.empty()).when(mockRunScheduleDao).loadRunSchedule(anyLong());
+
+        runScheduleManager.loadSchedule(1L);
     }
 
+
     private PipelineRunScheduleVO getRunScheduleVO(final RunScheduledAction action, final String cronExpression) {
-        PipelineRunScheduleVO runScheduleVO = new PipelineRunScheduleVO();
+        final PipelineRunScheduleVO runScheduleVO = new PipelineRunScheduleVO();
         runScheduleVO.setTimeZone(TIME_ZONE.getDisplayName());
         runScheduleVO.setCronExpression(cronExpression);
         runScheduleVO.setAction(action);
         return runScheduleVO;
     }
 
-    private void createPipelineRun(final Long runId, final Long pipelineId) {
+    private PipelineRun createPipelineRun(final Long runId, final Long pipelineId) {
         final AbstractCloudRegion cloudRegion = ObjectCreatorUtils.getDefaultAwsRegion();
-        regionDao.create(cloudRegion);
         PipelineRun run = ObjectCreatorUtils.createPipelineRun(runId, pipelineId, null, cloudRegion.getId());
         run.setStatus(TaskStatus.RUNNING);
         run.getInstance().setSpot(false);
-        pipelineRunDao.createPipelineRun(run);
+        return run;
     }
 
     private RunConfiguration createRunConfiguration() {
-        //create
-        RunConfigurationEntry entry =
-                ObjectCreatorUtils.createConfigEntry("entry", true, null);
+        final RunConfigurationEntry entry =
+                ObjectCreatorUtils.createConfigEntry(ENTRY_NAME, true, null);
 
-        RunConfiguration configuration =
+        final RunConfiguration configuration =
                 ObjectCreatorUtils.createConfiguration(TEST_NAME, null, null,
                         USER_OWNER, Collections.singletonList(entry));
-        return configurationDao.create(configuration);
+        return configuration;
     }
 
+    private RunSchedule createRunSchedule() {
```

Comment:
It seems like we can get rid of this method in advance to the overloaded version of the same method with several parameters.